src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -81,42 +81,48 @@ public class DocumentValidator : IDocumentValidator
+            try
             {
-                OriginalQuery = originalQuery ?? document.OriginalQuery,
-                Schema = schema,
-                Document = document,
-                TypeInfo = new TypeInfo(schema),
-                UserContext = userContext,
-                Inputs = inputs
-            };
+                context.OriginalQuery = originalQuery ?? document.OriginalQuery;
+                context.Schema = schema;
+                context.Document = document;
+                context.TypeInfo = new TypeInfo(schema);
+                context.UserContext = userContext;
+                context.Inputs = inputs;
 
-            List<INodeVisitor> visitors;
+                List<INodeVisitor> visitors;
 
-            if (useOnlyStandardRules)
-            {
-                // No async/await related allocations since all standard rules return completed tasks from ValidateAsync.
-                visitors = new List<INodeVisitor>();
-                foreach (var rule in (List<IValidationRule>)rules) // no iterator boxing
+                if (useOnlyStandardRules)
                 {
-                    var visitor = rule.ValidateAsync(context)?.Result;
-                    if (visitor != null)
-                        visitors.Add(visitor);
+                    // No async/await related allocations since all standard rules return completed tasks from ValidateAsync.
+                    visitors = new List<INodeVisitor>();
+                    foreach (var rule in (List<IValidationRule>)rules) // no iterator boxing
+                    {
+                        var visitor = rule.ValidateAsync(context)?.Result;
+                        if (visitor != null)
+                            visitors.Add(visitor);
+                    }
+                }
+                else
+                {
+                    var awaitedVisitors = rules.Select(x => x.ValidateAsync(context)).Where(x => x != null);
+                    visitors = (await Task.WhenAll(awaitedVisitors)).ToList();
                 }
-            }
-            else
-            {
-                var awaitedVisitors = rules.Select(x => x.ValidateAsync(context)).Where(x => x != null);
-                visitors = (await Task.WhenAll(awaitedVisitors)).ToList();
-            }   
 
-            visitors.Insert(0, context.TypeInfo);
+                visitors.Insert(0, context.TypeInfo);
 
-            new BasicVisitor(visitors).Visit(document, context);
+                new BasicVisitor(visitors).Visit(document, context);
 
-            return context.HasErrors
-                ? new ValidationResult(context.Errors)
-                : (IValidationResult)SuccessfullyValidatedResult.Instance;
+                return context.HasErrors
+                    ? new ValidationResult(context.Errors)
+                    : (IValidationResult)SuccessfullyValidatedResult.Instance;
```

Comment:
Suggest changing these three lines of code to:
```cs
                success = !context.HasErrors;

                return context.HasErrors
                    ? new ValidationResult(context.Errors)
                    : (IValidationResult)SuccessfullyValidatedResult.Instance;
```

Reply:
?
```c#
 success = !context.HasErrors;

                return success
                    ? (IValidationResult)SuccessfullyValidatedResult.Instance
                    : new ValidationResult(context.Errors);
```

Reply:
Looks good to me